packages/bindings/src/posts/models.rs

```diff
@@ -0,0 +1,342 @@
+    pub url: String,
+    /// Display value of the URL.
+    pub display_url: Option<String>,
+}
+
+/// Contains the details of entities parsed out of the post text.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct Entities {
+    /// Hashtag entities.
+    pub hashtags: Vec<TagEntity>,
+    /// Mention entities.
+    pub mentions: Vec<TagEntity>,
+    /// Url entities.
+    pub urls: Vec<UrlEntity>,
+}
+
+/// Contains the details of a post reference.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct PostReference {
+    /// Type of reference.
+    #[serde(rename = "type")]
+    pub ref_type: PostReferenceType,
+    /// Id of the referenced post.
+    pub post_id: Uint64,
+    /// Position of the reference inside the post's text. This should be used only
+    /// with the type set to ['PostReferenceType::Quote']
+    pub position: Option<Uint64>,
+}
+
+/// Represents the different types of [`PostReference`].
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum PostReferenceType {
+    /// No reference specified.
+    Unspecified,
+    /// Represents a reply to the specified post.
+    ReplayTo,
+    /// Represents a quote of the specified post.
+    Quote,
+    /// Represents a repost of the specified post.
+    Repost,
+}
+
+/// Contains the possible reply settings that a post can have.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ReplySetting {
+    /// No reply setting specified.
+    #[serde(rename = "REPLY_SETTING_UNSPECIFIED")]
+    Unspecified,
+    /// Everyone will be able to reply to this post.
+    #[serde(rename = "REPLY_SETTING_EVERYONE")]
+    Everyone,
+    /// Only followers of the author will be able to reply to this post.
+    #[serde(rename = "REPLY_SETTING_FOLLOWERS")]
+    Followers,
+    /// Only the author mutual followers will be able to reply to this post.
+    #[serde(rename = "REPLY_SETTING_MUTUAL")]
+    Mutual,
+    /// Only people mentioned inside this post will be able to reply.
+    #[serde(rename = "REPLY_SETTING_MENTIONS")]
+    Mentions,
+}
+
+/// Contains the data of a single post attachment.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct Attachment {
+    /// Id of the subspace inside which the post to which this attachment should be
+    /// connected is.
+    pub subspace_id: Uint64,
+    /// Id of the subspace section inside which the post to which this attachment
+    /// should be connected is.
+    pub section_id: u32,
+    /// Id of the post to which this attachment should be connected.
+    pub post_id: Uint64,
+    /// Id of this attachment.
+    pub id: u32,
+    /// Content of the attachment.
+    pub content: RawPostAttachment,
```

Comment:
We can use trait to replace `RawPostAttachment`, something like:
```rust
pub struct Attachment<a> {
    ....
    pub content: Box<dyn AttachmentContent<'a>>,
}

trait AttachmentContent<'a> : Serialize + Deserialize<'a> {};

struct MediaAttachment{
    ...
};
impl AttachmentContentfor for MediaAttachment;

struct PollAttachment{
    ...
}
impl AttachmentContent for PollAttachment;
```

Reply:
Nevermind, it seems no available to do that since trait can not be wrapped into a correct struct when deserializing.